### DISCUSSION TITLE
更新主线15图需要修改哪些内容才能实现

### DISCUSSION THREAD

**Discussion by guoh064** [LmeSzinc/AzurLaneAutoScript#3475] [General]
最近更新15图的issue有好几个。大部分都是要求先有周回自动的。也有看到用空文件占位先凑活用的。

15图的出击界面有4排舰队而不是3排，所以可能需要仿照FleetPreparation类写一个15图专用的类（毕竟也不知道16图会是个什么样子）
然后是非周回模式下的移动舰队的操作，可能也得在fleet.py里面新增函数来实现；
地图文件的话目前的map_extractor里面mystery项似乎对应15图里面的超级航母，但是map_extractor读不到超航和挡路舰队的位置，这个可能也得改。

目前想到要改的就这几样。欢迎补充。

**Comment by Arcohyp** [comment]
> 最近更新15图的issue有好几个。大部分都是要求先有周回自动的。也有看到用空文件占位先凑活用的。
> 
> 15图的出击界面有4排舰队而不是3排，所以可能需要仿照FleetPreparation类写一个15图专用的类（毕竟也不知道16图会是个什么样子） 然后是非周回模式下的移动舰队的操作，可能也得在fleet.py里面新增函数来实现； 地图文件的话目前的map_extractor里面mystery项似乎对应15图里面的超级航母，但是map_extractor读不到超航和挡路舰队的位置，这个可能也得改。
> 
> 目前想到要改的就这几样。欢迎补充。

@guoh064 我这边只知道周回15-1和非周回15-4[#3417 ](https://github.com/LmeSzinc/AzurLaneAutoScript/issues/3417)的情况，打捞后只挂过15-1周回。

15-1战斗逻辑和前面一样，但在空文件占位下存在其他问题：

在选择道中boss舰队逻辑因为使用的是 image_color_count 匹配，所以 _auto_search_set_click 会提示 “No active auto search setting found”，这个问题其实是匹配图片的时候没有设置 offset ，需要换一个匹配方式或者重写一个方法，再或者增加 assets(感觉没必要)。实际上即使有这个报错，依然会进入战斗，只不过是 fleet1_all_fleet2_standby 模式。

_当然，即使这样，单队绿海清15-1在现在的条件下应当是十分富裕的。如果只是老咸鱼刷心智，几乎不用改。_

提到的这个 FleetPreparation 对于1、2和潜艇队伍的选择是没有问题的，因为匹配时预先留下了 offset 。唯一需要实现的就是第四行的航空支援舰队，这块应该打开船坞筛选-降序-选中最高的三个，可能没法选择特定的舰娘。或者另一种解决方法是自己预先设定好，跑起来不会动那里的。

15-2应该类似15-1，而15-3的情况则烦请各位补充了。
